### DISCUSSION TITLE
Does jsign support generating a PKCS7 structure?

### DISCUSSION THREAD

**Discussion by littlell** [ebourg/jsign#156] [Q&A] [answered]
I just want to sign a digest string to a PKCS7 result, instead of signing a execute file like exe etc, thanks!

**Comment by ebourg** [comment, marked as answer]
For now Jsign is focused on code signing using Microsoft Authenticode. The scope of the project may be expanded to other file formats in the future (like JAR, PDF or even raw files), but it's not a priority.

**Comment by littlell** [comment]
> For now Jsign is focused on code signing using Microsoft Authenticode. The scope of the project may be expanded to other file formats in the future (like JAR, PDF or even raw files), but it's not a priority.

Thanks for your reply,  now we met a problem,  the case is:

A windows program calculate the target exe's message digest, like sha1 or sha256, and then send the digest to the saas java code signing service, so the java side accept the digest as a input parameter, not the entire exe file, so the java service need to calculate a PKCS7 result for the digest, and finally return the pkcs7 to the windows program, the windows program will append the PKCS7 result to the target exe file.

How can I use jsign for this case? thanks!

**Comment by ebourg** [comment]
I don't think you need Jsign in this case, you can directly use Bouncy Castle.

**Comment by littlell** [comment]
> I don't think you need Jsign in this case, you can directly use Bouncy Castle.

Yeah, I used Bouncy Castle to do the PKCS7, but we met error when we analyzed the P7 structure by a windows program, so I'm not sure whether it's different for the standard BC and jsign when making PKCS7 result.

thanks!

**Comment by ebourg** [comment]
What program do you use on Windows to check the signature?

**Comment by littlell** [comment]
> What program do you use on Windows to check the signature?

when parsing the P7 we met a crash，by debuging we found that the p7 lost some fields.

the verify tool is written by ourselves.

thanks

**Comment by littlell** [comment]
I just want to ask whether there is a difference between standard bc and jsign when calculate the p7
thanks

**Comment by ebourg** [comment]
Jsign uses Bouncy Castle to generate the PKCS#7 signature, but the way the file is hashed is specific to Authenticode.

**Comment by littlell** [comment]
> Jsign uses Bouncy Castle to generate the PKCS#7 signature, but the way the file is hashed is specific to Authenticode.

OK, thanks

The problem may come from the windows p7 check program
